File: Algorithms/234. Palindrome Linked List/golang/solution.recursive.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
var frontPointer *ListNode

func isPalindrome(head *ListNode) bool {
	frontPointer = head
	return recursivelyCheck(head)
}

func recursivelyCheck(currentNode *ListNode) bool {
	if currentNode == nil {
		return true
	}
	if !recursivelyCheck(currentNode.Next) {
		return false
	}
	if frontPointer.Val != currentNode.Val {
		return false
	}
	frontPointer = frontPointer.Next
	return true
}

File: Algorithms/234. Palindrome Linked List/php/solution.recursive.php
<?php

/**
 * Definition for a singly-linked list.
 * class ListNode {
 *     public $val = 0;
 *     public $next = null;
 *     function __construct($val = 0, $next = null) {
 *         $this->val = $val;
 *         $this->next = $next;
 *     }
 * }
 */
class Solution
{

    private $frontPointer = null;

    /**
     * @param ListNode $head
     * @return Boolean
     */
    function isPalindrome($head)
    {
        $this->frontPointer = $head;
        return $this->recursivelyCheck($head);
    }

    function recursivelyCheck($currentNode)
    {
        if (is_null($currentNode)) {
            return true;
        }
        if (!$this->recursivelyCheck($currentNode->next) || $currentNode->val != $this->frontPointer->val) {
            return false;
        }
        $this->frontPointer = $this->frontPointer->next;
        return true;
    }
}
